src/main.ts

```diff
@@ -0,0 +1,124 @@
+import os = require('os');
+
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import * as io from '@actions/io';
+import * as toolCache from '@actions/tool-cache';
+import * as rustCore from '@actions-rs/core';
+
+function getPlatformMatchingTarget(): string {
+    const platform = os.platform() as string;
+    switch (platform) {
+        case "linux":
+            return "x86_64-unknown-linux-gnu"
+        case "win32":
+            return "x86_64-pc-windows-msvc"
+        case "darwin":
+            return "x86_64-apple-darwin"
```

Comment:
I don't think GA supports that [yet](https://github.com/actions/runner-images/issues/5631)?

Reply:
As far as I know, all runners hosted by GitHub use x86_64 architecture. We might consider adding support for other platforms, but the questions are:

1. Does anyone really need this action running in such a specific environment like the self-hosted runner based on for example arm macOS?
2. How would we test the support for other platforms? It would require using a runner different from these provided by GitHub. I don't have access to any arm-based device at the moment (other then my phone :-p), so I'm not sure if I'm able to do it.

Reply:
If GitHub Actions doesn't support them then we don't have to add them now either. Let's make an issue to revisit this in the future and link to the upstream issue for ARM-based macOS and Linux builds.

In general, semver-checking is sensitive to the platform because conditional compilation allows [parameterizing based on the platform](https://doc.rust-lang.org/reference/conditional-compilation.html#target_arch). The generated rustdoc JSON represents the configured platform. Most crates don't have platform-specific differences in API, but some lower-level crates probably do and we ideally would support them: if semver-checking is hard on one platform, it's even harder when you might semver-break only "x86 on linux with avx2 enabled."

Reply:
Does the upstream cargo-semver-checks issue regarding ARM-based systems already exists? I can't find any.

Reply:
Feel free to open one and add the context from here.

It isn't clear what cargo-semver-checks could do in this case, other than document the limitation and advise users to generate rustdoc (or run the entire checking process) on the target they care about.

On the cargo-semver-checks side, a bigger limitation right now is the inability to specify features to check, so the target-specific issue has taken the back seat at the moment.

Reply:
Added the issue for future: https://github.com/obi1kenobi/cargo-semver-checks-action/issues/24.